db/updateItem.go

```diff
@@ -24,13 +21,18 @@ func (d *dynamoDB) UpdateItem(listID string, itemID string, newName string) (*da
 		panic("Items table name not set")
 	}
 
-	input := &dynamodb.PutItemInput{
-		Item:                itemToInsert,
-		TableName:           aws.String(tableName),
-		ConditionExpression: aws.String("attribute_exists(Id) AND attribute_exists(ListId)"),
+	fieldsToUpdate, updateExpression, expressionAttributeNames := getUpdateFields(newName, isCompleted)
+	input := &dynamodb.UpdateItemInput{
+		ExpressionAttributeValues: fieldsToUpdate,
+		Key:                       key,
+		TableName:                 aws.String(tableName),
+		UpdateExpression:          updateExpression,
+		ReturnValues:              aws.String("ALL_NEW"),
+		ExpressionAttributeNames:  expressionAttributeNames,
+		ConditionExpression:       aws.String("attribute_exists(Id)"),
 	}
 
-	_, err = d.session.PutItem(input)
```

Comment:
We used to just use PutItem and write over the top, but since we don't always want to update every field we now need to use UpdateItem:

https://docs.aws.amazon.com/sdk-for-go/api/service/dynamodb/#DynamoDB.UpdateItem